build/int.cloudbuild.yaml

```diff
@@ -28,8 +28,11 @@ steps:
       fi
 
       git fetch --unshallow
-      git diff origin/${_BASE_BRANCH} --name-only > _changed_files
-      sed 's/\/.*/\//' _changed_files > _changed_folders
+      git diff origin/main --name-only > _changed_files
+      sed 's:[^/]*$::' _changed_files > _changed_folders
```

Comment:
I don't think this worked as anticipated. 

From the output, this correctly detected the folders, but because of the trailing slashes, the check on line 38-44 didn't fire, and thus didn't exit early. 

We'd have to update the values on line 38 to adapt to this updated output. 

But additionally, the minimus change in this PR won't ever work, because since we're changing a file in build/ we should be running all tests anyway.

Reply:
Thanks @glasnt! I think originally this behavior was valuable to test the PR's "intended behavior", but I neglected to add a TODO to come back and fix when complete.